gandalf.js

```diff
@@ -3,122 +3,131 @@
  *
```

Comment:
SCRIPT-This logic might be clearer if you used a switch statement instead of multiple if-else statements.